src/rulesets/oas/index.json

```diff
@@ -342,11 +342,7 @@
       "description": "Parameter objects should have a `description`.",
       "recommended": false,
       "type": "style",
-      "given": "$..parameters[*]",
-      "when": {
-        "field": "@key",
-        "pattern": "^(?!.*(\\$ref))"
-      },
+      "given": "$..[?(@path === \"$['parameters']\" || (@.parameters && @parentProperty !== 'links'))]..[?(@.in)]",
```

Comment:
@P0lip, I'm kinda pulling my hair here. I can't find a rule that would satisfy OAS2 and OAS3. I was trying to create a rule that would eliminate `parameters` under `links` from the equation. I also wanted to make sure that even if we have:
```yaml
links:
  abc: 
    parameters:
      param:
         in: body
```
then we don't require `.description`. I don't know, but I feel like we might need more rules or a custom function.
WDYT?

Reply:
Ideally, Spectral would support an array of paths, so that you can supply a few json path expressions.
We could implement it as a part of 4.2 or 5.0.
For now, we could simply have 2 separate rules, one for oas2, and the other for oas3.
Once an array of paths can be provided, we can merge the rules.
Thoughts?

Reply:
We can support "string or array" and keep it as non breaking. The next release is likely to be 5.0 but supporting both is just a good idea either way for the sake of existing users. No BC-breaks-for-the-sake-of-it here. :)

Reply:
@lag-of-death OAS2 doesn't have links :p

Reply:
thanks @P0lip, I'll try with more than one rule for this now. @philsturgeon, "OAS2 doesn't have links", that's correct. As I was saying in: https://github.com/stoplightio/spectral/pull/515/files/1a6bac8bc7ee4a3063b1e4d61f8c2b163993bb6a#r320012847 "I can't find a rule that would satisfy OAS2 and OAS3". So this was an attempt to have just one rule. Didn't work out